Automattic/jetpack
# VideoPress: Expose purchases on client initial state

Fixes #27522.

Relates to #27524.

#### Changes proposed in this Pull Request:

* Exposed the site purchases list on the client initial state, dropping the need of a request to the My Jetpack purchases endpoint; the list is only loaded if the site is connected

#### Other information:


#### Jetpack product discussion



Technical details here: p1669037665069109-slack-C03TA48NSUX.

#### Does this pull request change what data or activity we track or use?



No.

#### Testing instructions:






* Open you browser console to inspect the client initial state and the network activity
* Go to `Jetpack > VideoPress`
* Notice that no requests are made to the `my-jetpack/v1/site/purchases` endpoint
* Inspect the value of the `jetpackVideoPressInitialState.initialState.purchases` var; for a connected site with an active subscription, it should look like this:

<img width="538" alt="Screen Shot 2022-11-21 at 15 55 16" src="https://user-images.githubusercontent.com/6760046/203137154-2c757f9e-ae2d-4977-abad-996c05fbe50a.png">

* You can also inspect the value of the `purchases` variable on the `videopress/media` store using the Redux debugger:

<img width="764" alt="Screen Shot 2022-11-21 at 15 51 03" src="https://user-images.githubusercontent.com/6760046/203136324-3ea0723e-af74-4d96-99b5-8b3fef1b8b34.png">

* Make sure the "Add new video" button is enabled if the site is connected and has an active subscription
